File: 71. Simplify Path/simplifyPath.js
/**
 * @param {string} path
 * @return {string}
 */
const simplifyPath = (path) => {
  const stack = [];
  const pathArr = path.split('/');

  for (let i = 0; i < pathArr.length; i++) {
    if (pathArr[i] === '..') {
      stack.pop()
    } else if (pathArr[i] && pathArr[i] !== '.') {
      stack.push(pathArr[i]);
    }
  }

  return '/' + stack.join('/');
};


console.log(simplifyPath('/home/')); // '/home'

File: 71. Simplify Path/simplifyPath.py
class Solution:
  def simplifyPath(self, path: str) -> str:
    stack = []
    for item in path.split('/'):
      if item == '..':
        if stack:
          stack.pop()
      elif item and item != '.':
        stack.append(item)

    return '/' + '/'.join(stack)


solution = Solution()
print('simplifyPath: ', solution.simplifyPath('/home/')) # /home
